File: Zuul_Java/Item.java
/*
 * All game items are objects of this class. The constructor initializes each 
 * object’s description, and an accessor method returns that description.
 * Author: Aahana Sapra
 * Date: 10/30/23
 */

public class Item {

	String description;
	
	/* Constructor 
	 */
	public Item(String newDescription) {
		description = newDescription;
	}
	
	/*
	 * Accessor method that returns the description of an item
	 */
	public String getDescription() {
		return description;
	}
}

File: Item.cpp
#include <iostream>
#include <cstring>
#include "Item.h"

using namespace std;

// Define constructor
Item::Item(char* description) {
  this->description = description;
}

// Define method to get item description
char* Item::getDescription() {
  return description;
}

// Define destructor
Item::~Item() {
  // do NOT delete description because it's a string literal
}
